Logs Watchdog Insights (#9787)
DataDog/documentation: content/en/logs/explorer/watchdog_insights.md

---
title: Watchdog Insights for Logs
kind: documentation
description: 'Get Insights on where to Start or Follow-Up your Investigations'
further_reading:
    - link: 'logs/explorer/side_panel'
      tag: 'Documentation'
      text: 'The log side panel'
    - link: 'logs/explorer/#list-of-logs'
      tag: 'Documentation'
      text: 'The list view of logs'
---

## Overview

With Watchdog Insights, find your way faster to the root cause of problems with contextual insights you receive as you explore logs. Watchdog Insights complements your expertise and instincts, giving nudges toward where you could focus your troubleshooting attention. 

<div class="alert alert-warning">

> Watchdog Insights for the Log Explorer is a beta feature, available for customers using Log Management. If you have feedback, contact <a href="https://docs.datadoghq.com/help">Datadog support</a>.

Reviewer: ```suggestion
Watchdog Insights for the Log Explorer is a beta feature, rolling out to customers using Log Management. If you have feedback, contact <a href="https://docs.datadoghq.com/help">Datadog support</a>.
```

</div>

On the following example, Watchog Insights highlights that `version:2.9.7` of a containerized Ruby application causes most of errors observed in a specific timerange. 

{{< img src="logs/explorer/watchdog_insights/overview.png" alt="Watchdog Insights" style="width:100%;" >}}

## Navigating Watchdog Insights 

> The Watchdog Insights banner comes on top of Log Explorer results. It runs in background, and eventually surfaces insights relevant to the current query. If you seek for troubleshooting inspiration:

Reviewer: ```suggestion
The Watchdog Insights banner appears in the Log Explorer results page and surfaces insights relevant to the current query:
```

 
{{< img src="logs/explorer/watchdog_insights/banner_collapsed.png" alt="Watchdog Insights banner (collapsed)" style="width:100%;" >}}

> * Either **expand** the Watchdog Insight banner, meant for a quick overview on insights,

Reviewer: ```suggestion
To see a quick overview of insights, expand the Watchdog Insight banner:
```

{{< img src="logs/explorer/watchdog_insights/banner_expanded.png" alt="Watchdog Insights banner (expanded)" style="width:100%;" >}}

> * Or click **view all** to open the Watchdog Insight side panel, meant to navigate insights displayed with richer summary, 

Reviewer: ```suggestion
For richer troubleshooting inspiration, click **View all** to open the Watchdog Insights side panel: 
```

{{< img src="logs/explorer/watchdog_insights/side_panel.png" alt="Watchdog Insights side panel" style="width:100%;" >}}

> Each insight comes with its own embedded interactions and Side Panel with detailed troubleshooting material. The insight interactions and side panel vary depending on the [sort of Watchdog Insight](#watchdog-insight-collection).

Reviewer: ```suggestion
Each insight comes with its own embedded interactions and side panel with detailed troubleshooting material. The insight interactions and side panel vary depending on the [type of Watchdog Insight](#watchdog-insights-collections).
```

> ## Watchdog Insights Collection

Reviewer: ```suggestion
## Watchdog Insights collections
```

### Error Outliers

> Error Outliers surface fields ([faceted][1] tags or attributes) that are characteristic of errors matching the current query.They are meant to accelerate troubleshooting by casting light on `key:value` that are statiscally over-represented among the errors, giving you hints on the possible root cause of problems.

Reviewer: ```suggestion
An Error Outliers insight shows fields (that is, [faceted tags or attributes][1]) that are characteristic of errors matching the current query. This casts light on `key:value` pairs that are statistically over-represented among the errors, giving you hints on the possible root cause of problems.
```

Typical error outliers examples are `env:staging`, `docker_image:acme:3.1` `http.useragent_details.browser.family:curl`.

> * The **banner card** version of the insight shows

Reviewer: ```suggestion
* The **banner card** version of the insight shows:
```

>   * the field
>   * to what proportion of errors and overall logs this field is contributing,

Reviewer: ```suggestion
  * The name of the field.
  * The proportion of errors and overall logs that this field is contributing.
```

{{< img src="logs/explorer/watchdog_insights/error_outlier_s_card.png" alt="Error Outlier card (S)" style="width:40%;" >}}

> * The **side panel card** of the insight shows, in addition,
> 
>   * the main [log pattern][2] for error logs having that field,

Reviewer: ```suggestion
* The **side panel card** of the insight shows, in addition:

  * The main [log pattern][2] for error logs that have the field.
```

{{< img src="logs/explorer/watchdog_insights/error_outlier_l_card.png" alt="Error Outlier card (L)" style="width:60%;" >}}

> * The **full side panel** of the card shows, in addition, 
> 
>   * the timeseries for error logs having that field,
>   * other fields often associated with those logs,
>   * the comprehensive list of [log patterns][2] for those logs

Reviewer: ```suggestion
* The **full side panel** of the card shows, in addition:

  * The timeseries for error logs that have the field.
  * Other fields that are often associated with those logs.
  * A comprehensive list of [log patterns][2] for those logs.
```
